prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/application/DeployApplicationTask.java

```diff
@@ -0,0 +1,132 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AllArgsConstructor
+public class DeployApplicationTask extends BaseWorkFlowTask {
+
+	public static final String KUBECONFIG = "kubeconfig";
+
+	public static final String MANIFESTS_PATH = "manifestsPath";
+
+	public static final String NAMESPACE = "namespace";
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key(KUBECONFIG).type(WorkParameterType.TEXT).optional(false)
+						.description("kubeconfig file of the target cluster in json format").build(),
+				WorkParameter.builder().key(MANIFESTS_PATH).type(WorkParameterType.TEXT).optional(false)
+						.description("The path to the manifests to deploy the application").build(),
+				WorkParameter.builder().key(NAMESPACE).type(WorkParameterType.TEXT).optional(false)
+						.description("The namespace in which the application should be deployed").build());
+	}
+
+	/**
+	 * Executes the task logic, based on the following steps:
+	 * <ol>
+	 * Verify parameters are present
+	 * </ol>
+	 * <ol>
+	 * Load the manifests
+	 * </ol>
+	 * <ol>
+	 * Load the kubeconfig file and prepare the client
+	 * </ol>
+	 * <ol>
+	 * Deploy the manifests
+	 * </ol>
+	 * @param workContext context in which this unit of work is being executed
+	 * @return a {@link WorkReport} with the result of the execution
+	 */
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		final String kubeconfig;
+		final String manifestsPath;
+		final String namespace;
+
+		// Verify parameters are present
+		try {
+			kubeconfig = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, KUBECONFIG);
+			manifestsPath = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, MANIFESTS_PATH);
+			namespace = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, NAMESPACE);
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		// Load the manifests
+		final Set<Path> manifests;
+		try {
+			try (Stream<Path> walk = Files.walk(Paths.get(manifestsPath))) {
+				manifests = walk.filter(Files::isRegularFile)
+						.filter(file -> file.toString().endsWith(".yaml") || file.toString().endsWith(".yml"))
+						.collect(Collectors.toSet());
+			}
+		}
+		catch (IOException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new RuntimeException(
+					"Failed to read manifest files from path %s with error %s".formatted(manifestsPath, e.getMessage()),
+					e));
+		}
+
+		if (manifests.isEmpty()) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new RuntimeException("No manifest files found in path %s".formatted(manifestsPath)));
+		}
+
+		// Load the kubeconfig file and prepare the client
+		try (OpenShiftClient osClient = createOpenShiftClient(kubeconfig)) {
+			manifests.forEach(manifest -> applyManifest(osClient, namespace, manifest));
+		}
+		catch (Exception e) {
```

Comment:
applyManifest throws a RuntimeException, no?

Reply:
yes, I'll add a specific case for it after I'll find out what will be thrown in the case of  a failure to create a resource or for reading a resource after it was created.
there is a metadata object returned by this call that needs to be analyzed and to store the router information on the context. this PR isn't complete yet.

Reply:
added tests to cover the various cases